common/__init__.py

```diff
@@ -220,11 +220,14 @@ def fix_unit(unit):
 class FormatNumberJsonEncoder(json.JSONEncoder):
     def encode(self, obj):
         def recursive_format_number(obj):
-            if isinstance(obj, (int, float)):
+            # in python, bools are a subclass of int, so we should check explicitly
+            # if obj is not a bool, otherwise it will be converted to a float…
+            if isinstance(obj, (int, float)) and not isinstance(obj, bool):
                 return float(f"{obj:.6g}")
```

Comment:
This is also where we could check for the number being exactly `0` and avoid formatting it as float I guess.